databroker/broker.py

```diff
@@ -0,0 +1,448 @@
+from __future__ import print_function
+import warnings
+import six  # noqa
+from collections import deque, defaultdict
+
+import tzlocal
+import logging
+import numbers
+from .core import (Header, _inspect_descriptor,
+                   get_events as _get_events,
+                   get_table as _get_table,
+                   restream as _restream,
+                   process as _process, Images)
+
+
+# Toolz and CyToolz have identical APIs -- same test suite, docstrings.
+try:
+    from cytoolz.dicttoolz import merge
+except ImportError:
+    from toolz.dicttoolz import merge
+
+
+try:
+    from functools import singledispatch
+except ImportError:
+    try:
+        # We are running on Python 2.6, 2.7, or 3.3
```

Comment:
> Bluesky is a huge dependency that changes quite often under the hood. 

I would push back on that. The core functionality of bluesky (run_engine.py) depends on Python, jsonschema, and super_state_machine, all of which are light dependencies. And that list has been stable since the summer. The dependency creep is all due to the callbacks, and I think we agree that future callback development will be moving outside of the bluesky package. They are packed together currently only for our convenience.